diff --git a/src/ccache/argprocessing.cpp b/src/ccache/argprocessing.cpp
index ae74e522..5bc06fb0 100644
--- a/src/ccache/argprocessing.cpp
+++ b/src/ccache/argprocessing.cpp
@@ -1757,6 +1757,13 @@ process_args(Context& ctx)
     compiler_args.push_back("/showIncludes");
   }
 
+  if (!config.run_second_cpp()) {
+    // We ignore some different diagnose behaviors here, which will cause
+    //  different result from compiling source directly or from intermediate
+    //  preprocessed files.
+    compiler_args.push_back("-Wno-error=address");
+  }
+
   return ProcessArgsResult{
     preprocessor_args,
     extra_args_to_hash,
