--- a/opt/brave.com/brave/brave-browser	2024-12-11 10:44:13.142435365 +0800
+++ b/opt/brave.com/brave/brave-browser	2024-12-11 10:46:23.533731267 +0800
@@ -26,14 +26,28 @@
 fi
 
 export CHROME_VERSION_EXTRA="stable"
 
 # We don't want bug-buddy intercepting our crashes. http://crbug.com/24120
 export GNOME_DISABLE_CRASH_DIALOG=SET_BY_GOOGLE_CHROME
 
+# https://aur.archlinux.org/cgit/aur.git/tree/brave-bin.sh?h=brave-bin
+# 31c435c4f47eb95177e1ac003282e4d90c126e5d
+XDG_CONFIG_HOME="${XDG_CONFIG_HOME:-"${HOME}/.config"}"
+CONF_FILE="${XDG_CONFIG_HOME}/brave-flags.conf"
+if test -f "${CONF_FILE}"; then
+  mapfile -t CONF_LIST < "${CONF_FILE}"
+fi
+
+for CONF_LINE in "${CONF_LIST[@]}"; do
+  if ! [[ "${CONF_LINE}" =~ ^[[:space:]]*(#|$) ]]; then
+    FLAG_LIST+=("${CONF_LINE}")
+  fi
+done
+
 # Sanitize std{in,out,err} because they'll be shared with untrusted child
 # processes (http://crbug.com/376567).
 exec < /dev/null
 exec > >(exec cat)
 exec 2> >(exec cat >&2)
 
-"$HERE/brave" "$@" || true
+"$HERE/brave" "${FLAG_LIST[@]}" "$@" || true
