diff --git a/rime_engine.c b/rime_engine.c
index 4e08960..b84ae02 100644
--- a/rime_engine.c
+++ b/rime_engine.c
@@ -504,7 +504,8 @@ ibus_rime_engine_process_key_event (IBusEngine *engine,
                                     guint       modifiers)
 {
   // ignore super key
-  if (modifiers & IBUS_SUPER_MASK) {
+  // @see ibus_engine_filter_key_event
+  if (modifiers & (IBUS_SUPER_MASK | IBUS_MOD4_MASK)) {
     return FALSE;
   }
   
